### PR TITLE
Prune null edges in Eval nodes

### PR DESCRIPTION
**Pull request by apaszke**
Recent autograd refactor has destabilized the tracer, because it has to handle NULL edges in the graph now (it reverted some of my previous changes). This should fix it (`word_language_model` seems to run just fine)

### CODE REVIEW

File: torch/csrc/autograd/functions/jit_closure.cpp [hunk 1]
```diff
@@ -681,6 +681,9 @@ AutogradClosure::AutogradClosure(const std::shared_ptr<MultiStageClosure>& desc,
   , desc(desc)
   , stage(stage) {
   auto & stage_desc = desc->stages[stage];
+  // TODO: WTF
```
Comment (ezyang) [inline torch/csrc/autograd/functions/jit_closure.cpp]: What's this?

Comment (apaszke) [reply torch/csrc/autograd/functions/jit_closure.cpp]: Lol. I forgot to investigate one thing :P

**Comment by ezyang** [comment]
LGTM. I added some comments, in my usual style. I'll revive patches from my other PR which we should also keep.

**Comment by ezyang** [comment]
(CI looks failed because we canceled it on the comment commits.)

**Comment by zdevito** [comment]
LGTM.

**Pull request merged**